docs/Formula-Cookbook.md

```diff
@@ -251,7 +251,36 @@ uses_from_macos "curl", since: :monterey
 
 Homebrew doesn’t package already-packaged language-specific libraries. These should be installed directly from `gem`/`cpan`/`pip` etc.
 
-If you're installing an application then use [`resource`](https://rubydoc.brew.sh/Formula#resource-class_method)s for all language-specific dependencies:
+### Ruby Gem Dependencies
+
+The preferred mechanism for installing gem dependencies is to use `bundler` with the upstream's `Gemfile.lock`. This requires the upstream checks in their `Gemfile.lock`, so if they don't, it's a good idea to file a bug and ask them to do so. Assuming they have one, this is as simple as:
+
+```ruby
+ENV["GEM_HOME"] = libexec
+system "bundle", "install"
+```
+
+From there, you can build and install the project itself:
+
+```ruby
+system "gem", "build", "<project>.gemspec"
+system "gem", "install", "--ignore-dependencies", "<project>-#{version}.gem"
+```
+
+And install any bins, and munge their shebang lines, with:
+
+```ruby
+bin.install libexec/"bin/<bin>"
+bin.env_script_all_files(libexec/"bin", GEM_HOME: ENV["GEM_HOME"])
+```
+
+### Python dependencies
+
+something about requirements.txt ?
```

Comment:
@Homebrew/maintainers can I get some help filling this in?

Reply:
Just noting that Python is a bit trickier as there isn't a standardized lock.
- requirements.txt isn't used everywhere and I've seen many projects that just manually modify it so not really at the same level as tooling generated equivalents (e.g. ruby Gemfile.lock, go go.mod+go.sum, rust Cargo.lock).
- Poetry-based projects have poetry.lock but Poetry isn't adopted everywhere.
- Our current approach of resources is probably the most applicable and able to result in reproducible builds.

Given our current infra/DSL, I think Python dependencies would be covered under following other cases.

Reply:
Updated to incorporate this in and restructure slightly.